src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -1,36 +1,146 @@
+
+import org.checkerframework.checker.units.qual.Speed;
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
-import edu.wpi.first.wpilibj2.command.StartEndCommand;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.maps.RobotMap;
 
-public class Lift extends SubsystemBase {
+/*
+ * SUB REVIEW Lift Subsystem What Does It Do? 1) Raises and lowers lift Manually
+ * using two different moters in sync during the last 30 seconds of the game 2)
+ * Uses ratchet mechanism to break
+ * 
+ * What Modes Does It Have? 1) Up (Extend) 2) Down (Retract) 3) Brake (Ratchet)
+ * 4) Hook onto Bar using d-clip 5) Go to Middle Bar Position 6) Go to Top Bar
+ * Position
+ * 
+ * What Interactions Does It Have With Other Subsystems? 1) None
+ * 
+ * How Is It Triggered? 1) User Input
+ * 
+ * Does It Store Any State? 1) isBraked 2) Limit switch values 3) Encoder values
+ * 
+ * Sensors? 1) Encoders 2) Limit Switches
+ */
 
-    private SendableSpeedController elevatorMotor;
+public class Lift extends SubsystemBase {
 
+    private PIDSpeedController elevatorMotor;
+    private ISolenoid elevatorBrake;
+    private IEncoder liftEncoder;
+    private BooleanSupplier upperLimitSwitch;
+    private BooleanSupplier lowerLimitSwitch;
     private static final double elevatorMotorSpeed = 1;
 
+    boolean isBraked = elevatorBrake.get();
+
+    // TODO Find a way to sync the elevatorLeft and elevatorRight motors
     public Lift(RobotMap.LiftMap map) {
         super();
         elevatorMotor = map.elevator();
+        liftEncoder = map.getLiftEncoder();
+        elevatorBrake = map.liftBrake();
+        upperLimitSwitch = map.upperLiftLimit();
+        lowerLimitSwitch = map.lowerLiftLimit();
     }
 
-    public CommandBase up() {
-        return new StartEndCommand(() -> {
-            elevatorMotor.set(-elevatorMotorSpeed);
+    public void liftSpeed(double speed) {
+        if (speed > 0) {
+            isBraked = false;
+            if (upperLimitSwitch.getAsBoolean()) {
+                speed = 0;
+            }
+        } else if (speed <= 0) {
+            isBraked = true;
+            if (lowerLimitSwitch.getAsBoolean()) {
+                speed = 0;
+            }
+        }
+        elevatorBrake.set(isBraked);
+        elevatorMotor.set(speed);
+
+    }
+
+    // TODO make these heights reflect where we actually want them to be
+    public enum liftHeights {
+        Top(52.25), Middle(63), Bottom(78.125);
+
+        private double iPosition;
+
+        private double minLiftHeight = 51;
+
+        // Returning an double to compare whether we're in the right place or not
+        public double value() {
+            return this.iPosition - minLiftHeight;
+        }
+
+        // Returning the level the lift is at (top middle or bottom)
+        private liftHeights(double iPosition) {
+            this.iPosition = iPosition;
```

Comment:
This seems highly unlikely to ever be true. I suggest adding some sort of range around `this.iPosition`.

Reply:
getPosition >= iPoint.value() - 5 && getPosition <= iPoint.value() + 5

This is the best way i have found but it looks really bad. is there a better way to do it?